cli/daemon/dash/dash.go

```diff
@@ -66,21 +67,42 @@ func (h *handler) Handle(ctx context.Context, reply jsonrpc2.Replier, r jsonrpc2
 			ID      string `json:"id"`
 			Name    string `json:"name"`
 			AppRoot string `json:"app_root"`
+			Offline bool   `json:"offline,omitempty"`
 		}
-		runs := h.run.ListRuns()
+
 		apps := []app{} // prevent marshalling as null
-		seen := make(map[string]bool)
-		for _, r := range runs {
-			id := r.App.PlatformOrLocalID()
-			name := r.App.PlatformID()
-			if name == "" {
-				name = filepath.Base(r.App.Root())
+
+		// Load all the apps we know about
+		allApp, err := h.apps.List()
```

Comment:
Do we ever delete from this list? Like if the user deletes the directory? We probably should

Reply:
Yes we do already do that:

```go
// List lists all known apps.
func (mgr *Manager) List() ([]*Instance, error) {
	roots, err := mgr.listRoots()
	if err != nil {
		return nil, err
	}

	var apps []*Instance
	for _, root := range roots {
		app, err := mgr.resolve(root)
		if errors.Is(err, fs.ErrNotExist) {
			log.Debug().Str("root", root).Msg("app no longer exists, skipping")
			// Delete the
			_, _ = mgr.db.Exec(`DELETE FROM app WHERE root = ?`, root)
			continue
		} else if err != nil {
			log.Error().Err(err).Str("root", root).Msg("unable to resolve app")
			continue
		}
		apps = append(apps, app)
	}

	return apps, nil
}
```